src/hu_stuff.c

```diff
@@ -1057,6 +1058,40 @@ static void HU_UpdateCrosshair(void)
     if (linetarget && !(linetarget->flags & MF_SHADOW))
     {
       crosshair.cr = colrngs[hud_crosshair_target_color];
+      
+      // [Alaux] Lock crosshair on linetarget
+      if (hud_crosshair_lockon && gamestate == wipegamestate)
+      {
+        // This is essentially an adaptation of the code that determines
+        // where sprites are drawn on the screen, i.e. R_ProjectSprite()
+
+        extern fixed_t fractionaltic, viewcos, viewsin, projection;
+        fixed_t interpx, interpy, interpz, tr_x, tr_y, xscale;
+
+        // [AM] Interpolate between current and last position, if prudent.
+        if (uncapped && linetarget->interp == true && leveltime > oldleveltime)
+        {
+          interpx = linetarget->oldx + FixedMul(linetarget->x - linetarget->oldx, fractionaltic);
+          interpy = linetarget->oldy + FixedMul(linetarget->y - linetarget->oldy, fractionaltic);
+          interpz = linetarget->oldz + FixedMul(linetarget->z - linetarget->oldz, fractionaltic);
+        }
+        else {
+          interpx = linetarget->x;
+          interpy = linetarget->y;
+          interpz = linetarget->z;
+        }
+
+        tr_x = interpx - viewx;
+        tr_y = interpy - viewy;
+        xscale = FixedDiv(projection, FixedMul(tr_x, viewcos)
```

Comment:
Just an idea, but instead of duplicating half of `R_ProjectSprite()` here, wouldn't it make sense to add a check to that function if the `mobj_t *thing` passed as function parameter is the same as `linetarget` and update the crosshair coordinates from there?

Reply:
That sounds good on paper, yes. I'll try it when I have time.